cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsClient.java

```diff
@@ -125,107 +125,98 @@ private AmazonEC2 createAccess(AwsCredentialView awsCredential, String regionNam
 
     @VisibleForTesting
     AmazonEC2 createAccessWithClientConfiguration(AwsCredentialView awsCredential, String regionName, ClientConfiguration clientConfiguration) {
-        return proxy(AmazonEC2Client.builder()
+        AmazonEC2ClientBuilder clientBuilder = AmazonEC2Client.builder()
                 .withCredentials(getCredentialProvider(awsCredential))
                 .withClientConfiguration(clientConfiguration)
-                .withRequestHandlers(new AwsTracingRequestHandler(tracer))
-                .withRegion(regionName)
-                .build(), awsCredential, regionName);
+                .withRequestHandlers(new AwsTracingRequestHandler(tracer));
+        awsEndpointProvider.setupEndpoint(clientBuilder, "ec2", regionName, awsCredential.isGovernmentCloudEnabled());
+        return proxy(clientBuilder.build(), awsCredential, regionName);
     }
 
     public AmazonCloudWatchClient createCloudWatchClient(AwsCredentialView awsCredential, String regionName) {
-        AmazonCloudWatch client = proxy(com.amazonaws.services.cloudwatch.AmazonCloudWatchClient.builder()
+        AmazonCloudWatchClientBuilder clientBuilder = com.amazonaws.services.cloudwatch.AmazonCloudWatchClient.builder()
                 .withClientConfiguration(getDefaultClientConfiguration())
                 .withCredentials(getCredentialProvider(awsCredential))
-                .withRequestHandlers(new AwsTracingRequestHandler(tracer))
-                .withRegion(regionName)
-                .build(), awsCredential, regionName);
-        return new AmazonCloudWatchClient(client);
+                .withRequestHandlers(new AwsTracingRequestHandler(tracer));
+        awsEndpointProvider.setupEndpoint(clientBuilder, "monitoring", regionName, awsCredential.isGovernmentCloudEnabled());
```

Comment:
It would be less error-prone to use constants instead of these strings, representing the AWS service types.